File: 0867-transpose-matrix/0867-transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        result = [[0 for _ in range(len(matrix))] for _ in range(len(matrix[0]))]
        for j in range(len(matrix)):
            for i in range(len(matrix[0])):
                result[i][j] = matrix[j][i]
                
        return result

File: 0867-transpose-matrix/0867-transpose-matrix.go
func transpose(matrix [][]int) [][]int {
    result := make([][]int, len(matrix[0]))
    for i := range matrix[0] {
        result[i] = make([]int, len(matrix))
    }

    for i, row := range matrix {
        for j, item := range row {
            result[j][i] = item
        }
    }

    return result
}
